elastic/elastic-agent
# Capture stdout/stderr of spawned components

## What does this PR do?



Captures the standard out and standard error of spawned components of the command runtime. Parses their output and sends them through the logger at the underlying level bypassing any log level setting for the Elastic Agent. This allows the each component to have a lower log level than the Elastic Agent itself (useful for debugging).

This at the moment does not handle logging from the any service runtime (aka. Endpoint Security). I will look into that in a later branch.

## Why is it important?



Provides a unified view of what is happening with the components that the Elastic Agent is running. Before that was hidden in files now a unified file provides all logs from the elastic-agent and the spawned components.

**Note**: Without this change no logs from the running components in 8.6 get sent to Fleet with monitoring turned on.

## Checklist




## How to test this PR locally



Run the elastic-agent locally with `elastic-agent run -e` and you will see log messages from the spawned components shown in the output.

## Related issues


- Relates #221
